src/adapter/src/coord/sequencer.rs

```diff
@@ -2717,18 +2801,81 @@ impl<S: Append + 'static> Coordinator<S> {
         Ok(send_immediate_rows(rows))
     }
 
-    async fn sequence_explain_timestamp(
+    fn sequence_explain_timestamp_begin(
         &mut self,
-        session: &Session,
+        tx: ClientTransmitter<ExecuteResponse>,
+        session: Session,
         plan: ExplainPlan,
-    ) -> Result<ExecuteResponse, AdapterError> {
+    ) {
         let ExplainPlan {
             raw_plan,
             explainee,
             format,
             ..
         } = plan;
 
+        let window_functions = self.catalog.system_config().window_functions();
+
+        let explainee_is_view = matches!(explainee, Explainee::Dataflow(_));
+        let decorrelated_plan = try_exec!(
+            raw_plan.optimize_and_lower(&OptimizerConfig {
+                qgm_optimizations: session.vars().qgm_optimizations(),
+                window_functions: window_functions || explainee_is_view,
+            }),
+            tx,
+            session
+        );
+        let optimized_plan =
+            try_exec!(self.view_optimizer.optimize(decorrelated_plan), tx, session);
```

Comment:
This change is just code movement and the addition of `try_exec` (See below for where is was moved from).